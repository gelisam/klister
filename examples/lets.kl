#lang kernel

(example (the Syntax (let (x 'foo) x)))

(example (flet (f (x) (case x [(true) 'a] [(false) 'b])) (f (true))))

(example (the Syntax (flet (f (x) 'a) (list-syntax ((f (true)) (f 4)) 'here))))

(example (the Syntax (let (f (lambda (x) 'a)) (list-syntax ((f (true)) (f 4)) 'here))))

(define id (lambda (x) x))

(example (the Syntax (id 'foo)))
(example (the Integer (id 45)))
