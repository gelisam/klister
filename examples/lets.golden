#[lets.kl:3.31-3.34]<foo> : Syntax
#[lets.kl:5.40-5.41]<a> : Syntax
#[lets.kl:7.72-7.76]<(a a)> : Syntax
#[lets.kl:9.80-9.84]<(a a)> : Syntax
#[lets.kl:13.27-13.30]<foo> : Syntax
45 : Integer
